pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -274,19 +274,23 @@ func (f *framework) QueueSortFunc() LessFunc {
 // cycle is aborted.
 func (f *framework) RunPrefilterPlugins(
 	pc *PluginContext, pod *v1.Pod) *Status {
-	for _, pl := range f.prefilterPlugins {
+	ctx, cancel := context.WithCancel(context.Background())
+	errCh := schedutil.NewErrorChannel()
+	workqueue.ParallelizeUntil(ctx, 16, len(f.prefilterPlugins), func(index int) {
+		pl := f.prefilterPlugins[index]
 		status := pl.Prefilter(pc, pod)
 		if !status.IsSuccess() {
-			if status.Code() == Unschedulable {
-				msg := fmt.Sprintf("rejected by %v at prefilter: %v", pl.Name(), status.Message())
-				klog.V(4).Infof(msg)
-				return NewStatus(status.Code(), msg)
-			}
-			msg := fmt.Sprintf("error while running %v prefilter plugin for pod %v: %v", pl.Name(), pod.Name, status.Message())
-			klog.Error(msg)
-			return NewStatus(Error, msg)
+			msg := fmt.Errorf("rejected by %v at prefilter: %v", pl.Name(), status.Message())
+			klog.V(4).Info(msg)
+			errCh.SendErrorWithCancel(msg, cancel)
+			return
 		}
+	})
+
+	if err := errCh.ReceiveError(); err != nil {
+		return NewStatus(Unschedulable, err.Error())
```

Comment:
It breaks the original status code handling logic IMO.

Reply:
Yes, the `ErrorChannel` does not permit send an `Status` :(   However it might be OK because the caller of the function only cares whether it succeeds or not.

Reply:
Addressed it. Add an `Error()` for `Status`, then `ErrorChannel` could sent it now.